### PR TITLE
Feature: Support multilingual - Add read multiple language

### PR DESCRIPTION
**Pull request by haddadzineddine**
## Description

This pull request adds a new branch called language_return_all that returns all existing result when the requested language value is set to "all". This feature was implemented based on the request described in issue https://github.com/agroportal/project-management/issues/307

## Changes Made / Todos

- [x] Added new branch called language_return_all .

- [x] Implemented logic to return all existing result when the requested language value is set to "all".
- [x] Set @language to the requested language [ ontoportal-lirmm/ontologies_linked_data/pull/80 ]
- [x] Support select multilanguages 688ec5e8bdcba6b1f3bfaca7e4c2fd15f299d924
- [x] Show the values in all languages with their corresponding language. [ agroportal/project-management/issues/307 ]
- [x] In @language return the available languages for the result not the requested languages https://github.com/ontoportal-lirmm/ontologies_linked_data/pull/80/commits/fad47b4efc051afb46fc0c942fe837c140da2353
- [x] Make it works for all attribues not only `synonyms` and `prefLabel`
- [x] Fix tests

## Screenshots

use case :  [localhost:9393/ontologies/INRAETHES/classes](http://localhost:9393/ontologies/INRAETHES/classes/http%3A%2F%2Fopendata.inrae.fr%2FthesaurusINRAE%2Fc_22817?display=synonym)

1. `requested_lang = all` 
image
2.  `requested_lang = fr` 
image
3.  `requested_lang = en` 
image
4.  `requested_lang = fr,en` 
image

## Resources & Informations

For the `requested_lang = all`, we need to show the values in all languages but also their corresponding language, like this 
image

For the case none languages :

image

So for your use case of synonyms, it needs to be like this: 

```json
# if attribute of type list
"synonym": {
"en": ["syn_en_1", "syn_en_2"],
"fr": ["syn_fr_1"]
}
# if attribute of type no-list
"prefLabel": {
"en": "label_en_1",
"fr": "label_fr_1"
}

```

The global  issue that describes this is here: https://github.com/agroportal/project-management/issues/307

## Reviewers
@syphax-bouazzouni

### CODE REVIEW

**Comment by syphax-bouazzouni** [comment]
Good starting block.
 
For the `requested_lang = all`, you need to show the values in all languages but also their corresponding language, like this 
image

So for your use case of synonyms, it needs to be like this: 

```json
synonym": {
"en": [syn_en_1, syn_en_2],
"fr": [syn_fr_1
}
```

The global  issue that describes this is here: https://github.com/agroportal/project-management/issues/307

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]
Good starting block.
 
For the `requested_lang = all`, you need to show the values in all languages but also their corresponding language, like this 
image

For the case none languages :

image

So for your use case of synonyms, it needs to be like this: 

```json
# if attribute of type list
"synonym": {
"en": ["syn_en_1", "syn_en_2"],
"fr": ["syn_fr_1"]
}
# if attribute of type no-list
"prefLabel": {
"en": "label_en_1",
"fr": "label_fr_1"
}

```

The global  issue that describes this is here: https://github.com/agroportal/project-management/issues/307

**Comment by haddadzineddine** [comment]
## Phase 2 

see : 

### Tests
use case :  [localhost:9393/ontologies/INRAETHES/classes](http://localhost:9393/ontologies/INRAETHES/classes/http%3A%2F%2Fopendata.inrae.fr%2FthesaurusINRAE%2Fc_22817?display=synonym)

1. `requested_lang = all` 
image
2.  `requested_lang = fr` 
image
3.  `requested_lang = en` 
image
4.  `requested_lang = fr,en` 
image

File: lib/goo/sparql/mixins/solution_lang_filter.rb [hunk 1]
```diff
@@ -6,53 +6,41 @@ class  LanguageFilter
         attr_reader :requested_lang, :unmapped, :objects_by_lang
 
         def initialize(requested_lang: nil, unmapped: false, list_attributes: [])
+          @attributes_to_translate = [:synonym, :prefLabel]
```
Comment (syphax-bouazzouni) [inline lib/goo/sparql/mixins/solution_lang_filter.rb]: need to translate all the attributes, not only :synonym and :prefLabel

File: lib/goo/sparql/mixins/solution_lang_filter.rb [hunk 2]
```diff
@@ -141,6 +144,10 @@ def fill_other_languages?
           @fill_other_languages
         end
 
+        def is_literal(object)
```
Comment (syphax-bouazzouni) [inline lib/goo/sparql/mixins/solution_lang_filter.rb]: in ruby, we write `literal?`

**Review by syphax-bouazzouni — CHANGES_REQUESTED** [review]

File: lib/goo/sparql/mixins/solution_lang_filter.rb (hunk 1 above)
Comment (haddadzineddine) [reply lib/goo/sparql/mixins/solution_lang_filter.rb]: is there a variable that contains all the attributes?

File: lib/goo/sparql/mixins/solution_lang_filter.rb (hunk 2 above)
Comment (haddadzineddine) [reply lib/goo/sparql/mixins/solution_lang_filter.rb]: 5f81e7b624af2b1bbb09f9434ff86111862fb4bc

File: lib/goo/sparql/mixins/solution_lang_filter.rb (hunk 1 above)
Comment (haddadzineddine) [reply lib/goo/sparql/mixins/solution_lang_filter.rb]: I got this error if i tried it with all variables 

image

**Comment by syphax-bouazzouni** [comment]
Change the language key to lowercase

File: lib/goo/sparql/mixins/solution_lang_filter.rb (hunk 1 above)
Comment (syphax-bouazzouni) [reply lib/goo/sparql/mixins/solution_lang_filter.rb]: fixed with https://github.com/ontoportal-lirmm/goo/pull/32/commits#:~:text=Fix%20the%20issue%20of%20undefined%20%27id%27%20of%20the%20language%20filter%20module

**Comment by syphax-bouazzouni** [comment]
Added:

- Fix the issue of undefined 'id' of the language filter module (https://github.com/ontoportal-lirmm/goo/pull/32/commits/3967febae703a659a09725a5c5ccd59c4c1ff9f7)
- Show literal attributes in the properties section (unmapped ones) if we requested all the languages (https://github.com/ontoportal-lirmm/goo/pull/32/commits/78e6420388f671aff68b226410d56e8887e39425)
- Use portal language by default in the language filter module (https://github.com/ontoportal-lirmm/goo/pull/32/commits/7e1410c2bc349bc3eacca6882e9386d6db0b9e53) 
Using the portal language by default fixed all the tests, because before by default it will show all the languages and it was not backward compatible so breaking all the tests.

**Comment by syphax-bouazzouni** [comment]
Final screenshot 
image

**Pull request merged**